src/smc-webapp/project_files.cjsx

```diff
@@ -1994,14 +1994,24 @@ exports.ProjectFiles = rclass ({name}) ->
                 style   = {error_style}
                 onClose = {=>@props.actions.setState(error:'')} />
 
-    render_not_public_error: ->
-        if @props.redux.getStore('account').is_logged_in()
-            <ErrorDisplay title="Directory is not public" error={"You are trying to access a non public project that you are not a collaborator on. You need to ask a collaborator of the project to add you."} />
+    render_not_collaborator_error: ->
```

Comment:
Name this something like `render_access_error` since it covers publicity and collaborator status.